webapp/src/ts/modules/messages/messages.component.ts

```diff
@@ -152,6 +162,14 @@ export class MessagesComponent implements OnInit, OnDestroy {
     return this.messageContactService
       .getList()
       .then((conversations = []) => {
+        // remove the lineage level that belongs to the offline logged in user
+        if (this.currentLevel) {
+          conversations.forEach(conversation => {
+            if (conversation.lineage) {
+              conversation.lineage = conversation.lineage.filter(level => level !== this.currentLevel);
```

Comment:
CHT allows creating multiple places with the exact same name. It's probably not common but it can happen. 
@elvisdorkenoo do we have access to the place's `_id`?

Reply:
When calculating the lineage for `tasks` whose breadcrumbs are pretty news, yes. For `messages and reports`, the lineage is just an array containing the name of the places, but there's always a way to get the `_id` . 
We can also come back to the solution to **simply remove the last element**, the only problem is that the lineage array sometimes come out with some` _null items as last elements,_` so we can **filter them out, and then remove the last element of the lineage array**.

Reply:
IMHO the safest/simplest thing to do would be to filter the null values like you suggested and then remove the last element _if it matches the `currentLevel`_.  If it does not match, I think we should do nothing.

Reply:
yeah agree